src/__tests__/test-app.provider.ts

```diff
@@ -27,13 +27,12 @@ export class TestAppProvider extends AppProvider {
     }
   }
 
-  protected getApp(
-    module: DynamicModule | TestingModule,
-  ): Promise<INestApplication> {
-    if (!(module instanceof TestingModule))
-      return Promise.reject(
-        `${module.constructor.name} is not a TestingModule`,
-      );
+  protected getApp(module: T): Promise<INestApplication> {
+    if (!(module instanceof TestingModule)) {
+      const name =
+        typeof module === 'function' ? module.constructor.name : module;
```

Comment:
What is the goal of this change? module might be an object for example. This would print

```
[object Object] is not a TestingModule
```

Reply:
We have to satisfy TypeScript. As [we are now using generics](https://github.com/safe-global/safe-client-gateway/pull/1060#discussion_r1469380455), `module` is `unknown`.

It's theoretically a `DynamicModule` so we could do something like this but I'd welcome your suggestion:

```ts
if (!(module instanceof TestingModule)) {
  const message = module instanceof DynamicModule
    ? `${module.constructor.name} is not a TestingModule`
    : 'Not a TestingModule'
  return Promise.reject(message);
}
return Promise.resolve(module.createNestApplication());
```

Reply:
I think this introduces complexity for an unlikely use-case that should only happen in a test environment. I'd say to keep it simple for now and reject the promise with something like "Provided module is not a TestingModule". Wdyt?

Reply:
Agreed. I simplified it in 9e246808069e9a68cd3f8d1eaa2ab20cce6c3e64.